jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -667,7 +706,7 @@ def mos_niriss_parser(app, data_dir, obs_label=""):
 
         # We then populate the table inside this context manager as _add_to_table
         # does operations that also trigger link manager updates.
-        _add_to_table(app, source_ids, "Source ID")
+        _add_to_table(app, source_ids, "Source Name")
```

Comment:
It bothers me a little that this is now "Name" instead of "ID" when the contents of the column might be simple integers (which seem more like IDs than names to me). Maybe there could be some code to decide between "Name"/"ID"? That could get messy internally though if other code refers to the column...personally I would default to "ID".

Reply:
Here's the dilemma I was faced with. I agree with you Ricky, that really what we're after here is some sort of "Identifier" that *identifies* the target, whatever that is, a word, a number, etc. The problem is `SOURCEID` is an ACTUAL KEYWORD used by the JWST pipeline, so I don't want to put `Source ID` because I don't want it to be confused with the actual keyword itself!

We could perhaps have it be "Target ID" or something, but I'm trying to avoid using the same terminology as the actual header keyword we're defaulting to. Maybe this is a PO design question (@PatrickOgle because you brought up the original idea, but also @camipacifici @orifox @larrybradley). Regardless, I think it should be the same for all parsers (regardless of instrument), and would be nicer for it to be uniform (i.e. NOT switch between Name/ID) if at all possible.

Reply:
> The problem is `SOURCEID` is an ACTUAL KEYWORD used by the JWST pipeline, so I don't want to put `Source ID` because I don't want it to be confused with the actual keyword itself!

I think it makes sense to have the `SOURCEID` value go in a column called `Source ID`, we should get some other opinions on whether other people find that confusing.

Reply:
@camipacifici @PatrickOgle @orifox any opinion here?

Reply:
Crazy ideas:

1. Have the column name be dynamic. If you are grabbing from "Source ID", that column will be "Source ID"; If you are grabbing from "TARGNAME", that column will be "TARGNAME". However, doing it like this, you cannot load two datasets with different headers.
2. Have one additional column that tells you which header keyword that source name is taken from. Name this new column something generic. This bloats the table a little.